metriq_gym/schema_validator.py

```diff
@@ -0,0 +1,30 @@
+import json
+import os
+from jsonschema import validate
+
+CURRENT_DIR = os.path.dirname(os.path.abspath(__file__))
+SCHEMA_DIR = "schemas"
+SCHEMA_MAPPING = {
+    "Quantum Volume": "quantum_volume.schema.json",
+}
+
+
```

Comment:
Here is (a slightly refactored) tweak of this file:

```py
import json
import os
from jsonschema import validate, ValidationError


CURRENT_DIR = os.path.dirname(os.path.abspath(__file__))
DEFAULT_SCHEMA_DIR = os.path.join(CURRENT_DIR, "schemas")

SCHEMA_MAPPING = {
    "Quantum Volume": "quantum_volume.schema.json",
}


def load_json_file(file_path: str) -> dict:
    """
    Load and parse a JSON file.
    Raises a ValueError if the file cannot be loaded or parsed.
    """
    try:
        with open(file_path, "r") as file:
            return json.load(file)
    except FileNotFoundError:
        raise ValueError(f"File not found: {file_path}")
    except json.JSONDecodeError as e:
        raise ValueError(f"Invalid JSON format in file '{file_path}': {e}")


def load_schema(benchmark_name: str, schema_dir: str = DEFAULT_SCHEMA_DIR) -> dict:
    """
    Load a JSON schema based on the benchmark name.
    Raises a ValueError if the benchmark is unsupported or the schema cannot be loaded.
    """
    schema_filename = SCHEMA_MAPPING.get(benchmark_name)
    if not schema_filename:
        raise ValueError(f"Unsupported benchmark name: {benchmark_name}")

    schema_path = os.path.join(schema_dir, schema_filename)
    return load_json_file(schema_path)


def validate_params(params: dict, schema_dir: str = DEFAULT_SCHEMA_DIR) -> None:
    """
    Validate parameters against the corresponding JSON schema.
    Raises a ValidationError if the parameters do not match the schema.
    """
    schema = load_schema(params.get("benchmark_name"), schema_dir)
    try:
        validate(instance=params, schema=schema)
    except ValidationError as e:
        raise ValueError(f"Validation failed: {e.message}")


def load_and_validate(file_path: str, schema_dir: str = DEFAULT_SCHEMA_DIR) -> dict:
    """
    Load parameters from a JSON file and validate them against the corresponding schema.
    Raises a ValueError if loading or validation fails.
    """
    params = load_json_file(file_path)
    validate_params(params, schema_dir)
    return params
```

An overview of these changes:

1. Adding type-hints
2. Adding additional error-handling
3. Adding docstrings
4. Common JSON loading contained in new function (`load_json_file`)
5. More validation

You can decide to use, scrap, or completely ignore if you disagree with these.

Reply:
I agree with all of those, but next time use the Github suggestion feature, so I can just click a button 😂

Reply:
Actually no, I don't agree with 5. I don't like that some ValidationError are now becoming ValueError.